src/QMCWaveFunctions/Fermion/MultiSlaterDetTableMethod.cpp

```diff
@@ -1002,6 +1002,40 @@ void MultiSlaterDetTableMethod::evaluateDerivativesWF(ParticleSet& P,
   evaluateMultiDiracDeterminantDerivativesWF(P, optvars, dlogpsi);
 }
 
+void MultiSlaterDetTableMethod::evaluateDerivativesWF_local(Vector<ValueType>& dlogpsi)
+{
+  evaluateDerivativesMSD(psi_ratio_to_ref_det_, dlogpsi);
+}
+
+void MultiSlaterDetTableMethod::detRatios(Vector<ValueType>& ratios)
+{
+  int det_id             = 0;
+  ValueType psiinv       = static_cast<ValueType>(PsiValue(1.0) / psi_ratio_to_ref_det_);
+  const auto& detValues0 = Dets[det_id]->getRatiosToRefDet();
+
+  // CI only for now
+  assert(!csf_data_);
+  ratios.resize(C->size());
+  ratios[0] = psiinv;
+  for (size_t i = 1; i < C->size(); i++)
+  {
+    ValueType cdet = psiinv * detValues0[(*C2node)[det_id][i]];
+    // assume that evaluateLog has been called in opt routine before
+    for (size_t id = 0; id < Dets.size(); id++)
+      if (id != det_id)
```

Comment:
single out `det0` makes the code much harder to read and it is unnecessary.
```
ValueType cdet = psiinv;
for (size_t id = 0; id < Dets.size(); id++)
  cdet *= Dets[id]->getRatiosToRefDet()[(*C2node)[id][i]];
```

Reply:
How is it unnecessary?  The rest of the existing multi-det code is already written this way, so it seemed like a design "feature".

Reply:
Not sure which place you were referring to. In other places, det_id depends on an argument or something but is hard-coded in your case. Thus, better to write the code without singling out `det_id=0`.